Long command is invisible after recalling from history

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

#### Symptoms
The terminal goes blank after recalling a command from history that is longer than the width of the terminal and then pressing the home key. See the video for a demo.

#### Reproduction
1. Open a new terminal
2. Type in a command that is longer than the terminal is wide (So that it wraps)
3. Run the command
4. Use the up arrow (To show the command again)
5. Press the home key (To go to the beginning of the command)

#### Observations
* The terminal still runs the invisible command if you press enter
* If you use the left arrow key to move the cursor to the first line and then press the home key, the bug doesn't happen
* The bug does not happen if Oh My Posh is disabled (Via commenting out the init command in the `~/.bashrc` file)

https://user-images.githubusercontent.com/38930607/217111778-5ea79031-d0c0-4cdd-93b3-e6ee1a17f2e1.mp4

### Theme

Theme: Agnoster
Command: `eval "$(oh-my-posh init bash --config ~/.poshthemes/agnoster.omp.json)"`

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

bash

### Log output

```Shell
Version: 14.2.1

Segments:

ConsoleTitle(false) -   0 ms - 
root(false)        -   0 ms - 
session(true)      -  26 ms -  wakef@lambda 
path(true)         -   1 ms -   
git(false)         -   0 ms - 
python(false)      -   0 ms - 
exit(false)        -   0 ms - 

Run duration: 29.98006ms

Cache path: /home/wakef/.cache/oh-my-posh

Config path: /home/wakef/.poshthemes/agnoster.omp.json

Logs:

[DEBUG] 15:17:54.031 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.031 shell.go:Getenv(XDG_CACHE_HOME) - 44.408µs
[TRACE] 15:17:54.031 shell_unix.go:CachePath() - 72.682µs
[DEBUG] 15:17:54.032 shell.go:Getenv:347 → /home/wakef/.poshthemes/agnoster.omp.json
[TRACE] 15:17:54.032 shell.go:Getenv(POSH_THEME) - 8.23µs
[TRACE] 15:17:54.032 shell.go:resolveConfigPath() - 15.319µs
[TRACE] 15:17:54.032 shell.go:Init() - 1.037637ms
[TRACE] 15:17:54.032 shell.go:Flags() - 124ns
[TRACE] 15:17:54.033 config.go:loadConfig() - 569.532µs
[TRACE] 15:17:54.033 shell.go:Flags() - 91ns
[DEBUG] 15:17:54.033 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.033 shell.go:Getenv(OMP_CACHE_DISABLED) - 4.524µs
[TRACE] 15:17:54.033 shell.go:Flags() - 52ns
[TRACE] 15:17:54.033 shell.go:GOOS() - 91ns
[DEBUG] 15:17:54.033 shell.go:Pwd:375 → NO DATA
[TRACE] 15:17:54.033 shell.go:Pwd() - 19.644µs
[TRACE] 15:17:54.033 shell.go:GOOS() - 86ns
[TRACE] 15:17:54.033 shell_unix.go:Root() - 471ns
[DEBUG] 15:17:54.033 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.033 shell.go:Pwd() - 7.583µs
[TRACE] 15:17:54.033 shell.go:GOOS() - 47ns
[DEBUG] 15:17:54.033 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.033 shell.go:Getenv(SSH_CONNECTION) - 3.782µs
[DEBUG] 15:17:54.033 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.033 shell.go:Getenv(SSH_CLIENT) - 3.539µs
[TRACE] 15:17:54.033 shell_unix.go:Root() - 307ns
[DEBUG] 15:17:54.058 shell.go:Shell:592 → process name: bash
[TRACE] 15:17:54.058 shell.go:Shell() - 25.352819ms
[TRACE] 15:17:54.058 shell.go:ErrorCode() - 289ns
[DEBUG] 15:17:54.058 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.058 shell.go:Getenv(WSL_DISTRO_NAME) - 8.789µs
[TRACE] 15:17:54.058 shell_unix.go:IsWsl() - 18.491µs
[DEBUG] 15:17:54.058 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.058 shell.go:Pwd() - 11.706µs
[TRACE] 15:17:54.058 shell.go:GOOS() - 193ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 197ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 110ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 113ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 108ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 108ns
[TRACE] 15:17:54.058 shell.go:GOOS() - 106ns
[DEBUG] 15:17:54.058 shell.go:User:496 → wakef
[TRACE] 15:17:54.058 shell.go:User() - 10.074µs
[DEBUG] 15:17:54.058 shell.go:Host:508 → lambda
[TRACE] 15:17:54.058 shell.go:Host() - 25.492µs
[TRACE] 15:17:54.058 shell.go:GOOS() - 203ns
[TRACE] 15:17:54.058 shell.go:TemplateCache() - 25.670739ms
[TRACE] 15:17:54.059 shell.go:TemplateCache() - 489ns
[TRACE] 15:17:54.059 shell.go:Shell() - 377ns
[DEBUG] 15:17:54.060 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.060 shell.go:Pwd() - 12.27µs
[TRACE] 15:17:54.060 shell.go:GOOS() - 88ns
[DEBUG] 15:17:54.060 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.060 shell.go:Pwd() - 9.269µs
[TRACE] 15:17:54.060 shell.go:Shell() - 189ns
[TRACE] 15:17:54.060 shell.go:Shell() - 175ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 88ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 185ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 165ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 88ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 89ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 87ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 84ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 84ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 78ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 80ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 85ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 79ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 82ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 78ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 79ns
[TRACE] 15:17:54.060 shell.go:GOOS() - 81ns
[TRACE] 15:17:54.060 shell.go:PathSeparator() - 78ns
[DEBUG] 15:17:54.060 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.060 shell.go:Pwd() - 7.193µs
[DEBUG] 15:17:54.060 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.060 shell.go:Getenv(WSL_DISTRO_NAME) - 8.017µs
[TRACE] 15:17:54.060 shell_unix.go:IsWsl() - 16.641µs
[TRACE] 15:17:54.060 shell.go:StackCount() - 232ns
[TRACE] 15:17:54.060 shell_unix.go:DirIsWritable(/home/wakef) - 12.992µs
[TRACE] 15:17:54.060 shell.go:TemplateCache() - 231ns
[TRACE] 15:17:54.061 shell.go:TemplateCache() - 196ns
[TRACE] 15:17:54.061 shell.go:Shell() - 136ns
[DEBUG] 15:17:54.061 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.061 shell.go:Pwd() - 7.828µs
[TRACE] 15:17:54.061 shell.go:GOOS() - 158ns
[DEBUG] 15:17:54.061 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.061 shell.go:Getenv(WSL_DISTRO_NAME) - 6.911µs
[TRACE] 15:17:54.061 shell_unix.go:IsWsl() - 14.059µs
[TRACE] 15:17:54.061 shell_unix.go:IsWsl2() - 20.565µs
[TRACE] 15:17:54.061 shell.go:GOOS() - 72ns
[DEBUG] 15:17:54.061 shell.go:CommandPath:547 → /usr/bin/git
[TRACE] 15:17:54.061 shell.go:CommandPath(git) - 46.833µs
[TRACE] 15:17:54.061 shell.go:HasCommand(git) - 54.15µs
[DEBUG] 15:17:54.061 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.061 shell.go:Pwd() - 6.413µs
[ERROR] 15:17:54.061 shell.go:HasParentFilePath:680 → stat /.git: no such file or directory
[TRACE] 15:17:54.061 shell.go:HasParentFilePath(.git) - 44.075µs
[DEBUG] 15:17:54.061 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.061 shell.go:Pwd() - 6.29µs
[TRACE] 15:17:54.061 shell.go:GOOS() - 78ns
[DEBUG] 15:17:54.061 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.061 shell.go:Pwd() - 6.376µs
[DEBUG] 15:17:54.061 shell.go:Pwd:355 → /home/wakef
[TRACE] 15:17:54.061 shell.go:Pwd() - 19.675µs
[TRACE] 15:17:54.061 shell.go:GOOS() - 146ns
[TRACE] 15:17:54.061 shell.go:ErrorCode() - 160ns
[TRACE] 15:17:54.061 shell.go:ErrorCode() - 74ns
[DEBUG] 15:17:54.061 shell.go:Getenv:347 → NO DATA
[TRACE] 15:17:54.061 shell.go:Getenv(XDG_CACHE_HOME) - 6.004µs
[TRACE] 15:17:54.061 shell_unix.go:CachePath() - 40.423µs
[TRACE] 15:17:54.061 shell.go:Flags() - 260ns
```

--- a/src/ansi/ansi_writer.go
+++ b/src/ansi/ansi_writer.go
@@ -232,13 +232,6 @@ func (w *Writer) ConsolePwd(pwdType, userName, hostName, pwd string) string {
 	}
 }
 
-func (w *Writer) ClearAfter() string {
-	if w.Plain {
-		return ""
-	}
-	return w.clearLine + w.clearBelow
-}
-
 func (w *Writer) FormatTitle(title string) string {
 	title = w.trimAnsi(title)
 	if w.Plain {

--- a/src/engine/engine.go
+++ b/src/engine/engine.go
@@ -153,10 +153,6 @@ func (e *Engine) getTitleTemplateText() string {
 }
 
 func (e *Engine) renderBlock(block *Block, cancelNewline bool) {
-	defer func() {
-		e.write(e.Writer.ClearAfter())
-	}()
-
 	// when in bash, for rprompt blocks we need to write plain
 	// and wrap in escaped mode or the prompt will not render correctly
 	if e.Env.Shell() == shell.BASH && block.Type == RPrompt {
@@ -374,7 +370,6 @@ func (e *Engine) PrintTooltip(tip string) string {
 			return ""
 		}
 		text, length := block.RenderSegments()
-		e.write(e.Writer.ClearAfter())
 		e.write(e.Writer.CarriageForward())
 		e.write(e.Writer.GetCursorForRightWrite(length, 0))
 		e.write(text)

